vllm/model_executor/sampling_metadata.py

```diff
@@ -2,12 +2,16 @@
 from typing import Dict, List, Optional, Tuple
 
 import torch
+import random
 
 from vllm.sampling_params import SamplingParams, SamplingType
 from vllm.sequence import SequenceData
 from vllm.utils import in_wsl, is_neuron
+from vllm.model_executor.layers.triton_kernel.sample import (
+    get_num_triton_sampler_splits)
 
 _SAMPLING_EPS = 1e-5
+_SEED_0_REPLACEMENT = 3403598558
 
 
 class SamplingMetadata:
```

Comment:
Could we skip any of the new operations in this class in the case that no seeds are in use? (which I expect would be very common).

Reply:
Good question - I think there are three considerations:
* Skipping seeds could bring a little better performance.
* Skipping seeds introduces more special cases (undesirable).
* Not skipping seeds allows for request-level reproducibility on the server side, which could be useful for debugging model behavior.

Aside from those, triton random operations require some sort of a seed, so generating one would be necessary regardless.